### PR TITLE
fix(HttpUtils): Use UTF-8 when forwarding request bodies to OTP

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [ ] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [ ] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Looking for advice on unit tests for these changes if practical.

This PR fixes the OTP request maker so that the body content uses UTF-8 encoding to support all international content.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/utils/HttpUtils.java [hunk 1]
```diff
@@ -102,26 +101,16 @@ public static HttpResponseValues httpRequestRawResponse(URI uri, int timeoutInSe
...
-                    HttpPut putRequest = new HttpPut(uri);
-                    putRequest.setEntity(new StringEntity(bodyContent));
-                    putRequest.setConfig(timeoutConfig);
-                    httpUriRequest = putRequest;
-                } catch (UnsupportedEncodingException e) {
-                    LOG.error("Unsupported encoding type", e);
-                    return null;
-                }
+                HttpPut putRequest = new HttpPut(uri);
+                putRequest.setEntity(new StringEntity(bodyContent, "UTF-8"));
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/HttpUtils.java]: Could use `StandardCharsets.UTF_8` instead.

**Review by br648 — APPROVED** [review]
The change is good. Not sure how to test this to be honest!

File: src/main/java/org/opentripplanner/middleware/utils/HttpUtils.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/utils/HttpUtils.java]: Updated in https://github.com/ibi-group/otp-middleware/pull/188/commits/7fdc4ae631ddacd1bea52f2eac6429f5b982a847, thanks!

**Comment by JymDyerIBI** [comment]
Agree, the change looks good. We don't have an `HttpUtilsTest` class at all, but I'll poke around and see how best to test charsets like this.

It looks like this can be exercised by adding something similar to `ApiTestUtils#makePostRequest()` and `#makePutRequest()` methods and exercising those from the unit tests.

**Comment by JymDyerIBI** [comment]
It looks like this can be exercised by adding `ApiTestUtils#makePostRequest()` and `#makePutRequest()` methods and exercising those from the unit tests.

**Comment by JymDyerIBI** [comment]
Digging further, I see that POST and PUT are exercised in `ApiUserFlowTest` and `ConnectedDataPlatformTest` unit tests, so I think that should suffice to show that these changes don't break anything.

**Review by JymDyerIBI — APPROVED** [review]
Looks good to me.

**Pull request merged**